contexts/WalletManagerContext.tsx

```diff
@@ -587,7 +595,7 @@ export function WalletManagerProvider({ children }: { children: ReactNode }) {
     }
 
     await saveWalletSettings(walletSettings);
-  };
+  }, []);
```

Comment:
Why using callback here? In addition, I think here are missing some dependencies.

Reply:
![image](https://github.com/user-attachments/assets/f496ee8d-c1b7-4f6d-ac3d-2ce8d25c03b7)
It becomes an issue, that switching network would not change the addresses. The image is in mainnet now but still have test prefix.

Reply:
I introduce the `useCallback` to break the dependency chain and then the rerenders.
But it was more than a hack than the correct solution.

Thanks for removing it ;)